Build fails on Clang 18

```
#######################################################################
# Compiling C source src/ballet/json/cJSON.c to build/native/clang/obj/ballet/json/cJSON.o
#######################################################################
mkdir -pv build/native/clang/obj/ballet/json/ && \
clang -isystem ./opt/include -DFD_LOG_UNCLEAN_EXIT=1 -DFD_HAS_BACKTRACE=0 -DFD_USING_CLANG=1 -Wno-address-of-packed-member -Wno-unused-command-line-argument -Wno-bitwise-instead-of-logical -Xclang -target-feature -Xclang +fast-vector-fsqrt -march=native -mtune=native -Werror -Wall -Wextra -Wpedantic -Wstrict-aliasing=2 -Wconversion -Wdouble-promotion -Wformat-security -Wimplicit-fallthrough -O3 -ffast-math -fno-associative-math -fno-reciprocal-math -DFD_HAS_OPTIMIZATION=1 -g  -fno-omit-frame-pointer -D_FORTIFY_SOURCE=2 -fPIC -Wl,-z,now -fstack-protector-strong -DFD_HAS_INT128=1 -DFD_HAS_DOUBLE=1 -DFD_HAS_ALLOCA=1 -DFD_HAS_THREADS=1 -DFD_HAS_OPENSSL=1 -DFD_HAS_X86=1 -DFD_HAS_SSE=1 -DFD_IS_X86_64=1 -DFD_HAS_AESNI=1 -D_XOPEN_SOURCE=700 -DFD_HAS_HOSTED=1 -pthread -DFD_HAS_THREADS=1 -DFD_HAS_ATOMIC=1 -mfpmath=sse -falign-functions=32 -DFD_HAS_UCONTEXT=1 -DFD_BUILD_INFO=\"build/native/clang/info\" -std=c17 -DFD_HAS_SECP256K1=1 -DFD_HAS_ZSTD=1 -DFD_HAS_ROCKSDB=1 -DROCKSDB_LITE=1 -c src/ballet/json/cJSON.c -o build/native/clang/obj/ballet/json/cJSON.o
src/ballet/json/cJSON.c:115:25: error: use of NaN via a macro is undefined behavior due to the currently enabled floating-point options [-Werror,-Wnan-infinity-disabled]
  115 |         return (double) NAN;
      |                         ^
src/ballet/json/cJSON.c:115:25: error: use of NaN is undefined behavior due to the currently enabled floating-point options [-Werror,-Wnan-infinity-disabled]
  115 |         return (double) NAN;
      |                         ^~~
/usr/include/math.h:98:16: note: expanded from macro 'NAN'
   98 | #  define NAN (__builtin_nanf (""))
      |                ^~~~~~~~~~~~~~~~~~~
src/ballet/json/cJSON.c:565:9: error: use of NaN is undefined behavior due to the currently enabled floating-point options [-Werror,-Wnan-infinity-disabled]
  565 |     if (isnan(d) || isinf(d))
      |         ^~~~~~~~
/usr/include/math.h:1011:20: note: expanded from macro 'isnan'
 1011 | #  define isnan(x) __builtin_isnan (x)
      |                    ^~~~~~~~~~~~~~~~~~~
src/ballet/json/cJSON.c:565:21: error: use of infinity is undefined behavior due to the currently enabled floating-point options [-Werror,-Wnan-infinity-disabled]
  565 |     if (isnan(d) || isinf(d))
      |                     ^~~~~~~~
/usr/include/math.h:1030:20: note: expanded from macro 'isinf'
 1030 | #  define isinf(x) __builtin_isinf_sign (x)
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~
4 errors generated.
make: *** [config/everything.mk:311: build/native/clang/obj/ballet/json/cJSON.o] Error 1
```

--- a/src/util/tmpl/test_sort.c
+++ b/src/util/tmpl/test_sort.c
@@ -203,12 +203,12 @@ main( int     argc,
     FD_TEST( sort_up_search_geq( NULL, 0UL, 3.0f )==0UL );
     float sorted[1024];
     for( ulong j=0UL; j<1024UL; j++ ) sorted[j] = (float)j;
-    FD_TEST( sort_up_search_geq( sorted, 1024UL, -INFINITY )==0UL );
+    FD_TEST( sort_up_search_geq( sorted, 1024.0f, -9999.0f )==0UL );
     for( ulong j=0UL; j<2048UL; j++ ) {
       float query = sorted[j/2] + (((float)(j&1UL))/2.0f);
-      FD_TEST( sort_up_search_geq( sorted, 1024UL, query )==j/2 );
+      FD_TEST( sort_up_search_geq( sorted, 1024.0f, query )==j/2 );
     }
-    FD_TEST( sort_up_search_geq( sorted, 1024UL, INFINITY )==1023UL );
+    FD_TEST( sort_up_search_geq( sorted, 1024.0f, 9999.0f )==1023UL );
   } while(0);
 
   fd_rng_delete( fd_rng_leave( rng ) );

